reflection/serverreflection.go

```diff
@@ -269,9 +269,41 @@ func (s *serverReflectionServer) allExtensionNumbersForType(st reflect.Type) ([]
 	return out, nil
 }
 
+// fileDescWithDependencies returns a slice of serialized fileDescriptors in
+// wire format ([]byte). The fileDescriptors will include fd and all the
+// transitive dependencies of fd with names not in sentFileDescriptors.
+func fileDescWithDependencies(fd *dpb.FileDescriptorProto, sentFileDescriptors map[string]struct{}) ([][]byte, error) {
+	r := [][]byte{}
+	queue := make([]*dpb.FileDescriptorProto, 0)
+	queue = append(queue, fd)
+	for len(queue) > 0 {
+		currentfd := queue[0]
+		queue = queue[1:]
+		if _, exists := sentFileDescriptors[fd.GetName()]; len(r) == 0 || !exists {
```

Comment:
The "recommended" (by the Go team) pattern for implementing sets-as-maps in go is:

```go
map[string /* or whatever */]bool
if sent := sentFileDescriptors[fd.GetName()]; len(r) == 0 || !sent {
}
```

Reply:
Made this change.